care/facility/api/viewsets/facility_users.py

```diff
@@ -35,12 +35,21 @@ class FacilityUserViewSet(GenericViewSet, mixins.ListModelMixin):
 
     def get_queryset(self):
         try:
+            search_fields = {
+                key: self.request.query_params.get(key)
+                for key in self.search_fields
+                if self.request.query_params.get(key)
+            }
             facility = Facility.objects.get(
                 external_id=self.kwargs.get("facility_external_id"),
             )
             queryset = facility.users.filter(
                 deleted=False,
             ).order_by("-last_login")
+            if search_fields:
+                queryset = queryset.filter(
+                    **{key: value for key, value in search_fields.items() if value}
+                )
```

Comment:
`search_fields` is handled by `drf_filters.SearchFilter` filter back-end for us. Wasn't it working?

ref: https://www.django-rest-framework.org/api-guide/filtering/#searchfilter

cc: @sainak

Reply:
> `search_fields` is handled by `drf_filters.SearchFilter` filter back-end for us. Wasn't it working?
> 
> ref: https://www.django-rest-framework.org/api-guide/filtering/#searchfilter
> 
> cc: @sainak

@rithviknishad It will work if `search_text` is passed in the query parameter from FE. The only issue is search for facility users is specified as username only (similar to users page), but this viewset supports username/first and last names.

For context, LinkedFacilityUsers calls the same route and the AutoComplete input supports search by name/username there.

Anyways, the primary reason why I added this is to have similar behavior as users;  so we can just pass in `username` as query parameter from FE.